### ISSUE TITLE
keras.ops.roll fails on eager tensors when axis is a list but succeeds on Keras Input

### ISSUE THREAD

**Issue by AGFACBNNR** [keras-team/keras#22541]
### Summary
`keras.ops.roll` behaves inconsistently between eager tensors and Keras symbolic tensors when `axis` is a list/tuple. With TensorFlow backend, the eager path raises `shift and axis must have the same size`, while the symbolic path accepts the same arguments.

### Reproduction
```python
import keras
import tensorflow as tf

x_eager = tf.constant([[1., 2.], [3., 4.]], dtype=tf.float32)
x_sym = keras.Input(shape=(2,))

print(keras.ops.roll(x_eager, 0, axis=[0, 1]))
print(keras.ops.roll(x_sym, 0, axis=[0, 1]))
```

### Expected behavior
Both eager and symbolic tensors should either:
- support `axis=[0, 1]` consistently, or
- raise the same validation error up front.

### Actual behavior
- **Eager tensor:** raises `INVALID_ARGUMENT: shift and axis must have the same size`
- **Keras Input:** returns a symbolic tensor with shape `(None, 2)`
